examples/add_person.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/protobuf/examples/tutorial"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+func promptForAddress(r io.Reader) (*tutorial.Person, error) {
+	rd := bufio.NewReader(r)
+	p := &tutorial.Person{}
+	fmt.Print("Enter person ID number: ")
+	if _, err := fmt.Fscan(rd, &p.Id); err != nil {
+		return p, err
+	}
+	if _, err := rd.ReadString('\n'); err != nil {
+		return p, err
+	}
+
+	fmt.Print("Enter name: ")
+	name, err := rd.ReadString('\n')
+	if err != nil {
+		return p, err
+	}
+	p.Name = strings.TrimSpace(name)
+
+	fmt.Print("Enter email address (blank for none): ")
+	email, err := rd.ReadString('\n')
+	if err != nil {
+		return p, err
+	}
+	p.Email = strings.TrimSpace(email)
+
+	for {
+		fmt.Print("Enter a phone number (or leave blank to finish): ")
+		phone, err := rd.ReadString('\n')
+		if err != nil {
+			return p, err
+		}
+		phone = strings.TrimSpace(phone)
+		if phone == "" {
+			break
+		}
+		pn := &tutorial.Person_PhoneNumber{}
+		pn.Number = phone
+
+		fmt.Print("Is this a mobile, home, or work phone? ")
+		ptype, err := rd.ReadString('\n')
+		if err != nil {
+			return p, err
+		}
+		ptype = strings.TrimSpace(ptype)
+		switch ptype {
+		case "mobile":
+			pn.Type = tutorial.Person_MOBILE
+		case "home":
+			pn.Type = tutorial.Person_HOME
+		case "work":
+			pn.Type = tutorial.Person_WORK
+		default:
+			fmt.Println("Unknown phone type.  Using default.")
+		}
+		p.Phones = append(p.Phones, pn)
+	}
+
+	return p, nil
+}
+
+// Main reads the entire address book from a file, adds one person based on
+// user input, then writes it back out to the same file.
+func main() {
+	if len(os.Args) != 2 {
+		fmt.Fprintf(os.Stderr, "Usage:  %s ADDRESS_BOOK_FILE\n", os.Args[0])
+		os.Exit(-1)
+	}
+	fname := os.Args[1]
+	book := &tutorial.AddressBook{}
+
+	// Read the existing address book.
+	in, err := ioutil.ReadFile(fname)
+	if err != nil {
+		if os.IsNotExist(err) {
+			fmt.Printf("%s: File not found.  Creating new file.\n", fname)
+		} else {
+			fmt.Fprintln(os.Stderr, "Error reading file: ", err.Error())
```

Comment:
You generally don't need to write `.Error()` or `.String()` when sending values to fmt, since that package will do it for you.